bot/exts/utilities/challenges.py

```diff
@@ -268,30 +275,29 @@ async def challenge(self, ctx: commands.Context, language: str = "python", *, qu
         `.challenge <language> <query>, <difficulty>` - Pulls a random challenge with the query provided,
         under that difficulty within the language's scope.
         """
-        if language.lower() not in SUPPORTED_LANGUAGES["stable"] + SUPPORTED_LANGUAGES["beta"]:
+        language = language.lower()
+        if language not in SUPPORTED_LANGUAGES["stable"] + SUPPORTED_LANGUAGES["beta"]:
             raise commands.BadArgument("This is not a recognized language on codewars.com!")
 
         get_kata_link = f"https://codewars.com/kata/search/{language}"
         params = {}
 
-        if language and not query:
-            level = f"-{choice([1, 2, 3, 4, 5, 6, 7, 8])}"
-            params["r[]"] = level
-        elif "," in query:
-            query_splitted = query.split("," if ", " not in query else ", ")
+        if query is not None:
+            if "," in query:
+                query_splitted = query.split("," if ", " not in query else ", ")
 
-            if len(query_splitted) > 2:
-                raise commands.BadArgument(
-                    "There can only be one comma within the query, separating the difficulty and the query itself."
-                )
+                if len(query_splitted) > 2:
```

Comment:
Shouldn't this be `!=`?

Reply:
No, since it's making sure that query_splitted isn't over the length of 2 (`python, 8` would be valid but `python, 8, let's break the bot` wouldn't be valid.)